webroot/js/categoriesTree.js

```diff
@@ -0,0 +1,32 @@
+(function(){
+    'use strict';
+
+    new AutocompleteBox("categories_tree/autocomplete", function(suggestion) {
+        return suggestion.name;
+    }, "parentName", "autocompletionParent");
+
+    new AutocompleteBox("categories_tree/autocomplete", function(suggestion) {
+        return suggestion.name;
+    }, "categoryName", "autocompletionCategory");
+
+    new AutocompleteBox("tags/autocomplete", function(suggestion) {
+        return suggestion.name + '(' + suggestion.nbrOfSentences + ')';
+    }, "tagName", "autocompletionTag");
+
+    angular
+        .module('app')
+        .controller('CategoriesTreeController', [
+            '$http', '$location', '$anchorScroll', CategoriesTreeController
+        ]);
+    
+    function CategoriesTreeController($http, $location, $anchorScroll) {
```

Comment:
You do not use any of the the Angular services so you do not need to inject them:
```js
angular.module('app').controller('CategoriesTreeController', [CategoriesTreeController]);

function CategoriesTreeController() { ... }
```